### PR TITLE
initial commit

### PR DESCRIPTION
**Pull request by humoflife**
Initial work in progress commit for a working version of Crossplane composition function-shell.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,52 @@
+REPO_URL="xpkg.upbound.io/upbound/function-shell"
```
Comment (jbw976) [inline Makefile:1]: this repo URL should now point to crossplane-contrib

File: README.md [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+from claim and other composition field values.
+
+Note: `function-shell` has to receive permissions in form of a rolebinding to
+read secrets and perform other actions that may be prohibited by default. Below
+is a `clusterrolebinding` that will work, but you should exercise appropriate caution when
+setting function permissions.
+
+```
+#!/bin/bash
+SA=$(kubectl -n upbound-system get sa -o name | grep function-shell | sed -e 's|serviceaccount\/|upbound-system:|g')
```
Comment (jbw976) [inline README.md:49]: The usage of `upbound-system` should probably be replaced with `crossplane-system` so crossplane community folks can easily use these examples, what do you think?

**Review by jbw976 — APPROVED** [review]
Great place to get started, looking forward to the further improvements you'll be making on this function! feel free to follow up on some of the small feedback when you make the next round of changes 😉

File: README.md (hunk 2 above)
Comment (humoflife) [reply README.md:49]: Updated. Will be in next PR.

File: Makefile (hunk 1 above)
Comment (humoflife) [reply Makefile:1]: Updated. Will be in next PR.

File: package/crossplane.yaml [hunk 3]
```diff
@@ -0,0 +1,16 @@
+---
+apiVersion: meta.pkg.crossplane.io/v1beta1
+kind: Function
+metadata:
+  name: function-shell
+  annotations:
+    meta.crossplane.io/maintainer: Markus Schweig <humoflife @ github>
+    meta.crossplane.io/source: xpkg.upbound.io/markuss/function-shell
```
[the remark this answers is not in the document]
Comment (humoflife) [reply package/crossplane.yaml:8]: Updated. Will be in next PR.

**Pull request merged**